src/Dto/Segments/ListSegments.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OneSignal\Dto\Segments;
+
+use OneSignal\Dto\AbstractDto;
+
+class ListSegments implements AbstractDto
+{
+    protected ?int $limit = null;
+
+    protected ?int $offset = null;
+
+    public function __construct(?int $limit = null, ?int $offset = null)
```

Comment:
```suggestion
    /**
     * @param int<0, 2147483648>|null $limit
     * @param int<0, 2147483648>|null $offset
     */
    public function __construct(?int $limit = null, ?int $offset = null)
```

Is there any reason to have constructor and setters? Maybe just `ListSegments::create()->limit(10)->offset(10)`